activesupport/lib/active_support/cache/redis_cache_store.rb

```diff
@@ -0,0 +1,389 @@
+# frozen_string_literal: true
+
+begin
+  require "redis"
+  require "redis/distributed"
+rescue LoadError
+  warn "The Redis cache store requires the redis gem. Please add it to your Gemfile."
+  raise
+end
+
+# Prefer the hiredis driver but don't require it.
+begin
+  require "redis/connection/hiredis"
+rescue LoadError
+end
+
+require "digest/md5"
+require "active_support/core_ext/marshal"
```

Comment:
For large-key truncation and for entry marshaling.

Reply:
md5 can be [problematic](https://github.com/rack/rack/issues/1048); while it is slow*er*, I think we should probably just use sha2.

Reply:
Agree. I used MD5 to match MemCacheStore. I'd like to flip them both.

Reply:
Switched to SHA2. MemCacheStore can catch up later.